### DISCUSSION TITLE
What foundational skills best for Payload?

### DISCUSSION THREAD

**Discussion by jhgfffxhjjhhh** [payloadcms/payload#2881] [Q&A] [answered]
If I have a foundation of HTML/CSS/JavaScript and wanted to build on that in a way that made Payload intuitive to use, what should I learn next and in what order? My initial thought from looking at the documentation is below but would love input from those who know better. 

node
Express 
Typescript
React
Mongodb

**Comment by DanRibbens** [comment, marked as answer]
Payload abstracts a lot so that if all you know is JS you can build cool things. I'll comment on each of the things you listed below.

node - not much to learn here really
Express - Pretty simple, just knowing how it works could be helpful for troubleshooting issues in a project or extending your app outside of Payload
Typescript - This is what you should focus on if you have only done JS, TS is a big improvement and will serve you well in your dev career.
React - Knowing React will be helpful if you want to customize the admin UI or build frontend websites using Next.js or plain React.
Mongodb - There isn't much to know here, Payload abstracts the DB so much that you can get by with very little knowledge here to be honest. Also other database support is coming very soon if you're wanting to use SQL.

**Reply by jhgfffxhjjhhh** [reply]
Thank you!

**Comment by Khanafsar55991** [comment]
Fine And Fair businesses information and support for me

**Reply by Khanafsar55991** [reply]
Thank you for your support
